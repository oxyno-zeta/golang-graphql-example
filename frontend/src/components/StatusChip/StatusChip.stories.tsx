import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import Stack from '@mui/material/Stack';
import StatusChip, { Props } from './StatusChip';

export default {
  title: 'Components/StatusChip',
  component: StatusChip,
  args: {
    label: 'Fake',
    // Cleaned because not planned to be used like this
    onDelete: undefined,
  },
} as Meta<typeof StatusChip>;

export const Playground: StoryFn<typeof StatusChip> = function C(args: Props) {
  return <StatusChip {...args} />;
};

export const Colors: StoryFn<typeof StatusChip> = function C(args: Props) {
  return (
    <Stack spacing={2}>
      <div>
        <StatusChip {...args} label="None" />
      </div>
      <div>
        <StatusChip {...args} color="default" label="default" />
      </div>
      <div>
        <StatusChip {...args} color="error" label="error" />
      </div>
      <div>
        <StatusChip {...args} color="info" label="info" />
      </div>
      <div>
        <StatusChip {...args} color="primary" label="primary" />
      </div>
      <div>
        <StatusChip {...args} color="secondary" label="secondary" />
      </div>
      <div>
        <StatusChip {...args} color="success" label="success" />
      </div>
      <div>
        <StatusChip {...args} color="warning" label="warning" />
      </div>
    </Stack>
  );
};
