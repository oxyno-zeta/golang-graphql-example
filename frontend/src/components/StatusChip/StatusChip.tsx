import React from 'react';
import Chip, { ChipProps } from '@mui/material/Chip';

type Props = ChipProps;

function StatusChip({ label, color, sx = {}, ...rest }: Props) {
  return (
    <Chip
      label={label}
      color={color}
      variant="outlined"
      size="small"
      sx={{
        ...sx,
        borderRadius: '5px',
        backgroundColor: (theme) =>
          color === 'default' || !color ? theme.palette.action.selected : `${theme.palette[color].main}60`,
      }}
      {...rest}
    />
  );
}

export default StatusChip;
