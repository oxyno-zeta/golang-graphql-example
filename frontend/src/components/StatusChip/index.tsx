import StatusChip, { Props as StatusChipProps } from './StatusChip';

export default StatusChip;
export type { StatusChipProps };
