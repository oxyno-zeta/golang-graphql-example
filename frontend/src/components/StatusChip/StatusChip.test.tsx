import React from 'react';
import { render } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import StatusChip from './StatusChip';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('StatusChip', () => {
  it('should display label with default color (no value)', async () => {
    const { container } = render(<StatusChip label="fake-label" />);
    // Now find text
    expect(container).toHaveTextContent('fake-label');
    expect(container.firstChild).toHaveClass('MuiChip-colorDefault');
    expect(container.firstChild).toHaveClass('MuiChip-outlinedDefault');
    expect(container).toMatchSnapshot();
  });

  it('should display label with default color', async () => {
    const { container } = render(<StatusChip label="fake-label" color="default" />);
    // Now find text
    expect(container).toHaveTextContent('fake-label');
    expect(container.firstChild).toHaveClass('MuiChip-colorDefault');
    expect(container.firstChild).toHaveClass('MuiChip-outlinedDefault');
    expect(container).toMatchSnapshot();
  });

  it('should display label with error color', async () => {
    const { container } = render(<StatusChip label="fake-label" color="error" />);
    // Now find text
    expect(container).toHaveTextContent('fake-label');
    expect(container.firstChild).toHaveClass('MuiChip-colorError');
    expect(container.firstChild).toHaveClass('MuiChip-outlinedError');
    expect(container).toMatchSnapshot();
  });
});
