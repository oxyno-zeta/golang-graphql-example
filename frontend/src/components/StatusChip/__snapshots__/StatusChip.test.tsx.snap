// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusChip should display label with default color (no value) 1`] = `
<div>
  <div
    class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1a1j1t6-MuiChip-root"
  >
    <span
      class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
    >
      fake-label
    </span>
  </div>
</div>
`;

exports[`StatusChip should display label with default color 1`] = `
<div>
  <div
    class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1a1j1t6-MuiChip-root"
  >
    <span
      class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
    >
      fake-label
    </span>
  </div>
</div>
`;

exports[`StatusChip should display label with error color 1`] = `
<div>
  <div
    class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorError MuiChip-outlinedError css-kg4ko1-MuiChip-root"
  >
    <span
      class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
    >
      fake-label
    </span>
  </div>
</div>
`;
